app/src/main/java/net/squanchy/service/firestore/FirestoreDbService.kt

```diff
@@ -155,4 +156,21 @@ class FirestoreDbService(private val db: FirebaseFirestore) {
             subscriber.setCancellable { registration.remove() }
         }
     }
+
+    fun tracks(): Observable<List<FirestoreTrack>> {
+        return Observable.create { subscriber ->
+            val registration = db.collection("tracks")
+                .addSnapshotListener { snapshot, exception ->
+                    if (exception != null && subscriber.isDisposed.not()) {
+                        subscriber.onError(exception)
+                        return@addSnapshotListener
+                    }
+                    subscriber.onNext(snapshot.documents.map { trackSnapshot ->
+                        trackSnapshot.toObject(FirestoreTrack::class.java).apply { id = trackSnapshot.id } // TODO should be done in the backend
```

Comment:
Since we don't have a view which injects the ID, we do it manually here. We should get rid of this and create a view on the backend instead.

Reply:
Open an issue 👅

Reply:
Updated squanchy-dev/squanchy-firebase#8